### PR TITLE
[NFC][Doc][RDF] Add "Efficient analysis in Python" section

### PR DESCRIPTION
**Pull request by stwunsch**

### CODE REVIEW

File: tree/dataframe/src/RDataFrame.cxx [hunk 1]
```diff
@@ -381,6 +382,78 @@ ROOT::EnableImplicitMT();
 ~~~
 Simple as that. More details are given [below](#parallel-execution).
 
+
+##  <a name="python"></a>Efficient analysis in Python
+
+You can use `RDataFrame` in Python due to the dynamic C++/Python translation of PyROOT. In general, the interface
+is the same than for C++, a simple example follows.
```
Comment (eguiraud) [inline tree/dataframe/src/RDataFrame.cxx:389]: typo: `same as`

File: tree/dataframe/src/RDataFrame.cxx [hunk 2]
```diff
@@ -381,6 +382,78 @@ ROOT::EnableImplicitMT();
...
+sum = df.Filter("x > 10").Sum("y")
+print(sum.GetValue())
+~~~
+
+### Simple usage of efficient C++ code in Python
+
+To perform more complex operations in the `RDataFrame` graph, e.g., in `Filter` and `Define` nodes, which don't
+fit into a simple expression string, you can just-in-time compile such functions directly in the Python script
+via the C++ interpreter cling. This approach has the advantage that you get the efficiency of compiled C++ code
+combined with the convenient workflow of a Python script. See the following snipplet for an example of how to
```
Comment (eguiraud) [inline tree/dataframe/src/RDataFrame.cxx:402]: typo: snipplet -> snippet

File: tree/dataframe/src/RDataFrame.cxx [hunk 3]
```diff
@@ -381,6 +382,78 @@ ROOT::EnableImplicitMT();
...
+ROOT.gSystem.Load("path/to/myLibrary.so") # Library with the myFilter function
+ROOT.gInterpreter.Declare('#include "myLibrary.h"') # Header with the definition of the myFilter function
+df = ROOT.RDataFrame("myTree", "myFile.root")
+sum = df.Filter("myFilter(x)").Sum("y")
+print(sum.GetValue())
+~~~
+
+### Just-in-time compilation of Python callables with numba
+
+We also offer to compile Python callables with fundamental types and arrays thereof using numba and then
```
Comment (eguiraud) [inline tree/dataframe/src/RDataFrame.cxx:430]: "We also offer to" -> "ROOT also offers the option to"

File: tree/dataframe/src/RDataFrame.cxx [hunk 4]
```diff
@@ -381,6 +382,78 @@ ROOT::EnableImplicitMT();
...
+using the function in `RDataFrame` from C++. The workflow requires the Python packages `numba` and `cffi`
+to be installed. See the following snipplet for a simple example or the full tutorial [here](pyroot004__NumbaDeclare_8py.html).
```
Comment (eguiraud) [inline tree/dataframe/src/RDataFrame.cxx:432]: snipplet -> snippet

File: tree/dataframe/src/RDataFrame.cxx (hunk 2 above)
Comment (stwunsch) [reply tree/dataframe/src/RDataFrame.cxx:402]: I just learnt that snipplet is not an English word :D Thanks!

**Comment by stwunsch** [comment]
Fixed and rebased!

**Pull request merged**